### PR TITLE
Admin validation requests

### PR DESCRIPTION
**Pull request by Khalid-Abed**

### CODE REVIEW

File: app/Http/Controllers/CareerLevelController.php [hunk 1]
```diff
@@ -74,11 +77,15 @@ public function edit(CareerLevel $careerLevel)
...
-            'name' =>$request->name
-        ]);
-        return redirect()->route('careerLevels.index')
+        if (CareerLevel::where('name', $request->name)->where('id', '!=', $careerLevel->id)->count() == 0) {
+            CareerLevel::where('id', $careerLevel->id)->update(['name' => $request->name]);
+             return redirect()->route('careerLevels.index')
                         ->with(session()->flash('success','careerLevel is Updated successfully .'));
+        }
+        else{
+            $err['name']='This is already exist';
```
Comment (ahmed-fawzy-metwally) [inline app/Http/Controllers/CareerLevelController.php:86]: Rename it to
This career level already exists.

File: app/Http/Controllers/CountryController.php [hunk 2]
```diff
@@ -78,11 +81,14 @@ public function edit(Country $country)
...
-        $country->update([
-            'name' =>$request->name
-        ]);
-        return redirect()->route('countries.index')
+        if (Country::where('name', $request->name)->where('id', '!=', $country->id)->count() == 0) {
+            Country::where('id', $country->id)->update(['name' => $request->name]);
+             return redirect()->route('countries.index')
                         ->with(session()->flash('success','Country is Updated successfully .'));
+        }else{
+            $err['name']='This is already exist';
```
Comment (ahmed-fawzy-metwally) [inline app/Http/Controllers/CountryController.php:89]: Rename it to
This country already exists.

File: app/Http/Controllers/GuestController.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+use App\Models\Job;
+use App\Models\JobType;
+use App\Models\Application;
+
+class GuestController extends Controller
+{
+    public function mainPage(){
+        $career=['5.png','1.jpg','2.jpg','3.jpg','4.jpg','6.jpg','2.jpg'];
+        $category=['1.jpg','2.jpg','3.jpg','4.jpg','5.jpg','2.jpg'];
+        $city=['1.jpg','2.jpg','3.jpg','4.jpg','5.jpg'];
```
Comment (ahmed-fawzy-metwally) [inline app/Http/Controllers/GuestController.php:22]: Rename ```$career``` to be ```$careerImages```
and the rest variables as it
***
Rename these images to have a meaningful name.
as ```City1.png```

File: app/Http/Controllers/GuestController.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+              return view('guest.mainPage')->with('jobs',Job::orderBy('created_at', 'DESC')->get())
```
Comment (ahmed-fawzy-metwally) [inline app/Http/Controllers/GuestController.php:23]: Are you use jobs in this view?

File: app/Http/Controllers/IndustryCategoryController.php [hunk 5]
```diff
@@ -71,9 +74,15 @@ public function edit(IndustryCategory $industryCategory)
...
-        $industryCategory->update($request->all());
-        return redirect(route("industry-categories.index", $industryCategory))
-            ->with(session()->flash('success','Industry category is Updated successfully .'));;
+        if (IndustryCategory::where('name', $request->name)->where('id', '!=', $industryCategory->id)->count() == 0) {
+            IndustryCategory::where('id', $industryCategory->id)->update(['name' => $request->name]);
+             return redirect()->route('industry-categories.index')
+                        ->with(session()->flash('success','Industry category is Updated successfully .'));
+        }
+        else{
+            $err['name']='This is already exist';
```
Comment (ahmed-fawzy-metwally) [inline app/Http/Controllers/IndustryCategoryController.php:83]: Rename it to
This category already exists.

File: app/Http/Controllers/LanguageController.php [hunk 6]
```diff
@@ -75,9 +78,15 @@ public function edit(Language $language)
...
-        $language->update($request->all());
-        return redirect(route('languages.index'))
-            ->with(session()->flash('success','Language is updated successfully .'));;
+        if (Language::where('name', $request->name)->where('id', '!=', $language->id)->count() == 0) {
+            Language::where('id', $language->id)->update(['name' => $request->name]);
+             return redirect()->route('languages.index')
+                        ->with(session()->flash('success','language is Updated successfully .'));
+        }
+        else{
+            $err['name']='This is already exist';
```
Comment (ahmed-fawzy-metwally) [inline app/Http/Controllers/LanguageController.php:87]: Rename it to
This language already exists.

File: app/Http/Controllers/CareerLevelController.php (hunk 1 above)
Comment (Khalid-Abed) [reply app/Http/Controllers/CareerLevelController.php:86]: done

File: app/Http/Controllers/CountryController.php (hunk 2 above)
Comment (Khalid-Abed) [reply app/Http/Controllers/CountryController.php:89]: done

File: app/Http/Controllers/GuestController.php (hunk 3 above)
Comment (Khalid-Abed) [reply app/Http/Controllers/GuestController.php:22]: done

File: app/Http/Controllers/GuestController.php (hunk 4 above)
Comment (Khalid-Abed) [reply app/Http/Controllers/GuestController.php:23]: sure, latest Job

File: app/Http/Controllers/IndustryCategoryController.php (hunk 5 above)
Comment (Khalid-Abed) [reply app/Http/Controllers/IndustryCategoryController.php:83]: done

File: app/Http/Requests/CareerLevelRequest.php [hunk 7]
```diff
@@ -24,7 +24,7 @@ public function authorize()
     public function rules()
     {
         return [
-            'name' => 'required|unique:career_levels|string'
+            'name' => 'required|string|min:3|regex:/^([a-zA-Z]+)(\s[a-zA-Z]+)*$/'
```
Comment (ahmed-fawzy-metwally) [inline app/Http/Requests/CareerLevelRequest.php:27]: Add ```- ``` as allowed characeter
as some jobs contain this character for example Full-stack developer

File: app/Http/Controllers/LanguageController.php (hunk 6 above)
Comment (Khalid-Abed) [reply app/Http/Controllers/LanguageController.php:87]: done

File: public/css/user/mainPage.css [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+    .search {
+        margin: 100px auto;
+    }
+    #interval{
+        height: 550px !important;
+    }
+    .search-button{
+        /* position: absolute;
+        top: 6px;
+        right: 30px; */
```
Comment (ahmed-fawzy-metwally) [inline public/css/user/mainPage.css:16]: Remove the commented lines.

File: public/css/user/mainPage.css [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+}
+.nav-font{
+    font-size: large;
+}
+.search{
+    margin: 200px auto;
+}
+.index-search-form {
+    position: relative;
+    /* display: inline-flex; */
```
Comment (ahmed-fawzy-metwally) [inline public/css/user/mainPage.css:38]: Remove this line.

File: public/css/user/mainPage.css [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+.career-level:hover{
+    opacity: 0.8 ;
+}
+.text-bottom{
+    bottom:5px;
+}
+.jobs-by-category
+{
+
+}
```
Comment (ahmed-fawzy-metwally) [inline public/css/user/mainPage.css:72]: Remove this class.

File: resources/views/guest/mainPage.blade.php [hunk 11]
```diff
@@ -0,0 +1,247 @@
...
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>Document</title>
+    <link rel="stylesheet" href="{{ asset('css/fontaswesme-all.min.css') }}">
+    <link rel="stylesheet" href="{{ asset("css/app.css") }}">
+    <link rel="stylesheet" href="{{asset('css/user/mainPage.css')}}">
+</head>
+<body class="bg-white">
+    <div class="container-fluid" id="interval" style="height: 658px; background: url({{asset('guest/category/13.jpg')}});background-repeat: no-repeat; background-size: cover;">
```
Comment (ahmed-fawzy-metwally) [inline resources/views/guest/mainPage.blade.php:13]: Don't use inline-style

**Pull request closed without merging**